src/components/v5/common/ActionSidebar/partials/MultiSigSidebar/partials/MultiSigWidget/partials/ApprovalStep/ApprovalStep.tsx

```diff
@@ -0,0 +1,333 @@
+                    isTall
+                  />
+                </div>
+              )}
+              {hasRejectionVotes && (
+                <div
+                  className={clsx('flex items-center gap-2.5', {
+                    'mt-1': hasApprovalVotes,
+                  })}
+                >
+                  {hasApprovalVotes && hasRejectionVotes && (
+                    <span className="text-4">
+                      {formatText({ id: 'multiSig.approval.rejections' })}
+                    </span>
+                  )}
+                  <ProgressBar
+                    progress={rejectionProgress}
+                    max={threshold}
+                    additionalText={formatText(
+                      {
+                        id: 'multiSig.approval.additionalText',
+                      },
+                      {
+                        threshold,
+                      },
+                    )}
+                    className="ml-[0.125rem] !text-xs"
+                    isTall
+                  />
+                </div>
+              )}
+            </div>
+            {isOwner && allSignatures.length > 5 && (
+              <div className="mt-2">
+                <StatusText
+                  status={StatusTypes.Info}
+                  textClassName="text-4 text-gray-900"
+                  iconAlignment="top"
+                  iconSize={16}
+                  iconClassName="text-gray-500"
+                >
+                  {formatText({
+                    id: 'multiSig.approval.creator.rejection',
+                  })}
+                </StatusText>
+              </div>
+            )}
+            {showWeekInfo && (
+              <div className="mt-2">
+                <StatusText
+                  status={StatusTypes.Info}
+                  textClassName="text-4 text-gray-900"
+                  iconAlignment="top"
+                  iconSize={16}
+                  iconClassName="text-gray-500"
+                >
+                  {formatText({
+                    id: 'multiSig.approval.week.rejection',
+                  })}
+                </StatusText>
+              </div>
+            )}
+          </>
+        ),
+        textClassName: 'text-4 text-gray-900',
+        iconAlignment: 'top',
+        iconSize: 16,
+        iconClassName: 'text-gray-500',
+      }}
+      sections={[
+        {
+          key: 'signatories',
+          content: (
+            <div>
+              <Tooltip
+                placement="bottom-start"
+                offset={[0, 4]}
+                tooltipContent={formatText({ id: 'multiSig.approval.tooltip' })}
+              >
+                <h5 className="mb-2 text-1">
+                  {formatText({ id: 'multiSig.approval.title' })}
+                </h5>
+              </Tooltip>
+              <Signees
+                signees={
+                  allSignatures as unknown as MultiSigUserSignatureFragment[]
```

Comment:
since there is limited data here, what if we create a type called `MultiSigSignee` and we just give it the `userAddress`, `profile` (or even better just the avatar and display name) and the vote? this way you won't need to typecast at all